src/Spice86.Core/Emulator/LoadableFile/Bios/BiosLoader.cs

```diff
@@ -43,11 +43,11 @@ public BiosLoader(IMemory memory, State state, ILoggerService loggerService) : b
     /// <param name="file">The path to the BIOS file to load.</param>
     /// <param name="arguments">Ignored for BIOS files.</param>
     /// <returns>The loaded BIOS file as a byte array.</returns>
-    public override byte[] LoadFile(string file, string? arguments) {
+    internal override (ushort CodeSegment, byte[] FileContent) LoadFile(string file, string? arguments) {
         byte[] bios = ReadFile(file);
         uint physicalStartAddress = MemoryUtils.ToPhysicalAddress(CodeSegment, 0);
         _memory.LoadData(physicalStartAddress, bios);
         SetEntryPoint(CodeSegment, CodeOffset);
-        return bios;
+        return (CodeSegment, bios);
```

Comment:
I would return the segmented address pointing to the entry point from which the caller can extract the segment

Reply:
I changed strategy, we are using a DOS Process Manager class now. :)